### PR TITLE
First pass of integration of sharing into redux-cozy-client

### PR DESCRIPTION
**Pull request by goldoraf**
All sharing features have not been migrated into redux-cozy-client yet, more PRs to come ;)

### CODE REVIEW

**Comment by goldoraf** [comment]
[(2) [PHO] Changer le layout d'un album partagé](https://trello.com/c/Vjx9MiEn/470-2-pho-changer-le-layout-dun-album-partag%C3%A9)

**Comment by goldoraf** [comment]
[(2) [PHO] Changer le layout d'un album partagé](https://trello.com/c/Vjx9MiEn/470-2-pho-changer-le-layout-dun-album-partag%C3%A9)

File: src/lib/redux-cozy-client/CozyAPI.js [hunk 1]
```diff
@@ -149,6 +152,21 @@ export default class CozyAPI {
...
+      await cozy.client.fetchJSON('GET', `/permissions/doctype/${doctype}/${sharingType}`)
+
+    const byMe = await fetchPermissions(doctype, SHARED_WITH_OTHERS)
+    const byLink = await fetchPermissions(doctype, SHARED_BY_LINK)
+    const withMe = await fetchPermissions(doctype, SHARED_WITH_ME)
+
+    return { byMe, byLink, withMe }
+  }
+
+  async fetchSharing (id) {
```
Comment (enguerran) [inline src/lib/redux-cozy-client/CozyAPI.js]: No need for `async` here

File: src/lib/redux-cozy-client/CozyAPI.js [hunk 2]
```diff
@@ -149,6 +152,21 @@ export default class CozyAPI {
     await cozy.client.data.removeReferencedFiles(normalized, ids)
     return ids
   }
+
+  async fetchSharingPermissions (doctype) {
+    const fetchPermissions = async (doctype, sharingType) =>
+      await cozy.client.fetchJSON('GET', `/permissions/doctype/${doctype}/${sharingType}`)
```
Comment (enguerran) [inline src/lib/redux-cozy-client/CozyAPI.js]: No need for `await` here

File: src/lib/redux-cozy-client/CozyClient.js [hunk 3]
```diff
@@ -48,6 +44,20 @@ export default class CozyClient {
...
+    const permissions = await this.api.fetchSharingPermissions(doctype)
+    const sharingIds = [
+      ...permissions.byMe.map(p => p.attributes.source_id),
+      ...permissions.withMe.map(p => p.attributes.source_id)
+    ]
+    const sharings = await Promise.all(sharingIds.map(id => this.api.fetchSharing(id)))
+    return { permissions, sharings }
+  }
+
+  async fetchFile (id) {
```
Comment (enguerran) [inline src/lib/redux-cozy-client/CozyClient.js]: No need for `async` here

File: src/lib/redux-cozy-client/CozyClient.js [hunk 4]
```diff
@@ -48,6 +44,20 @@ export default class CozyClient {
...
+    return this.api.fetchFile(id)
+  }
+
   async createFile (file, dirID) {
```
Comment (enguerran) [inline src/lib/redux-cozy-client/CozyClient.js]: No need for `async` here

File: src/lib/redux-cozy-client/slices/sharings.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+const getSharing = (state, id) => state.cozy.sharings.documents.find(s => s.attributes.sharing_id === id)
+
+export const getSharingDetails = (state, doc) => {
+  const perms = getDoctypePermissions(state, doc._type)
+  const byMe = perms.byMe.find(p => p.attributes.permissions['rule0'].values.indexOf(doc._id) !== -1)
+  const withMe = perms.withMe.find(p => p.attributes.permissions['rule0'].values.indexOf(doc._id) !== -1)
+  const byLink = perms.byLink.find(p => p.attributes.permissions['collection'].values.indexOf(doc._id) !== -1)
+  const sharer = withMe !== undefined
+    ? {
+      name: 'John Doe',
```
Comment (enguerran) [inline src/lib/redux-cozy-client/slices/sharings.js]: Who is __John Doe__?

**Pull request merged**